sdk/ml/azure-ai-ml/azure/ai/ml/operations/_operation_orchestrator.py

```diff
@@ -141,7 +142,7 @@ def get_asset_arm_id(
                 # CLI strips off azureml: in the schema, appending it back as required by backend
                 if (
                     azureml_type == "environments"
-                    and asset.startswith(CURATED_ENV_PREFIX)
+                    and (asset.startswith(CURATED_ENV_PREFIX) or re.match(REGISTRY_VERSION_PATTERN, f"azureml:{asset}"))
```

Comment:
How about removing this whole line# 145?

Reply:
I'm not sure, but also hesitant to do, I don't want to accidentally miss any e2e scenarios if this was required.
Will assume that, w/e is here is needed, and only patch the bug we know about.